axelrod/tests/unit/test_prober.py

```diff
@@ -99,6 +99,79 @@ def test_strategy(self):
         self.responses_test([D, C, C, D, C], [C, D, C, C], [C])
 
 
+class TestProber4(TestPlayer):
+
+    name = "Prober 4"
+    player = axelrod.Prober4
+    expected_classifier = {
+        'stochastic': False,
+        'memory_depth': 1,
+        'makes_use_of': set(),
+        'long_run_time': False,
+        'inspects_source': False,
+        'manipulates_source': False,
+        'manipulates_state': False
+    }
+    initial_sequence = [
+        C, C, D, C, D, D, D, C, C, D, C, D, C, C, D, C, D, D, C, D
+    ]
+    cooperation_pool = [C] * 5
+
+    def test_initial_strategy(self):
+        """Starts by playing CCDCDDDCCDCDCCDCDDCD."""
+        self.responses_test([], [], self.initial_sequence)
+
+    def test_strategy(self):
+        # After playing the initial sequence defects forever
+        # if the absolute difference in the number of retaliating
+        # and provocative defections of the opponent is smaller or equal to 2
+
+        provocative_histories = [
+            [C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [C, D, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [C, D, C, D, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [C, C, D, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [C, C, D, C, D, C, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [D, D, D, D, D, D, D, D, D, D, D, D, D, D, D, D, D, D, D, D],
+        ]
+
+        history1 = self.initial_sequence
+        responses = [D] * 10
+        for history2 in provocative_histories:
+            self.responses_test(history1, history2, responses)
+
+        # Otherwise cooperates for 5 rounds
+        unprovocative_histories = [
+            [C, C, D, C, D, D, D, C, C, D, C, D, C, C, D, C, D, D, C, D],
+            [D, D, C, D, C, C, C, D, D, C, D, C, D, D, C, D, C, C, D, C],
+            [C, C, D, C, D, D, C, C, C, C, C, C, C, C, C, C, C, C, C, C],
+            [C, C, D, C, D, D, C, C, D, C, C, C, C, C, C, D, D, D, C, C],
+            [C, C, C, C, D, D, C, C, D, C, C, D, D, C, D, C, D, C, C, C],
+        ]
+
+        history1 = self.initial_sequence
+        responses = self.cooperation_pool
+        for history2 in unprovocative_histories:
+            self.responses_test(history1, history2, responses)
+
+        # and plays like TFT afterwards
+            history1 += self.cooperation_pool
+            history2 += self.cooperation_pool
+            self.responses_test(history1, history2, [C])
+
+            history1 += [C]
+            history2 += [D]
+            self.responses_test(history1, history2, [D])
+
+            history1 += [D]
+            history2 += [C]
+            self.responses_test(history1, history2, [C])
+
+            history1 += [C]
+            history2 += [D]
+            self.responses_test(history1, history2, [D])
+
+
```

Comment:
Can we test that `turned_defector` is being set correctly? `responses_test` can take an `attrs` parameter (a dictionary) that checks the state of internal variables on a player.